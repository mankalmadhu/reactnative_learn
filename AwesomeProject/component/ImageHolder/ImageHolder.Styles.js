import {StyleSheet} from 'react-native';

module.exports  = StyleSheet.create({
  dimensions : {width: 193, height: 110}
});