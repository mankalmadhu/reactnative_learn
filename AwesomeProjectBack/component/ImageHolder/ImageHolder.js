import React, { Component } from 'react';
import { AppRegistry, View } from 'react-native';
import PropTypes from 'prop-types';
import styles from './ImageHolder.Styles';

export default class ImageHolder extends Component {
  render() {
    return (
      <image src={this.props.pick} style={styles.dimensions}/>
    );
  }
};

ImageHolder.propTypes = {
  pick : PropTypes.string.isRequired
};
