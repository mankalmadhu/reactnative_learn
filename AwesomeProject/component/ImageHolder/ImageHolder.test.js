import 'react-native';
import React from 'react';
import ImageHolder from './ImageHolder';

import Renderer from 'react-test-renderer';

test("Image Render" , () => {

   const imageRenderer = Renderer.create(
       <ImageHolder pick={'https://upload.wikimedia.org/wikipedia/commons/d/de/Bananavarieties.jpg'}/>
   ).toJSON();

   expect(imageRenderer).toMatchSnapshot();

});