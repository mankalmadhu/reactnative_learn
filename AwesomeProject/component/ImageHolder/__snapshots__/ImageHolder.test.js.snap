// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Render 1`] = `
<image
  src="https://upload.wikimedia.org/wikipedia/commons/d/de/Bananavarieties.jpg"
  style={
    Object {
      "height": 110,
      "width": 193,
    }
  }
/>
`;
